mitiq/executor/executor.py

```diff
@@ -147,6 +147,36 @@ def evaluate(
                 "Expected observable to be hermitian. Continue with caution."
             )
 
+        # Check executor and observable compatability with type hinting
+        # If FloatLike is specified as a return and observable is used
+        if self._executor_return_type in FloatLike and observable is not None:
+            # Type hinted as FloatLike and observable passed
+            if self._executor_return_type is not None:
+                raise ValueError(
+                    "When using a float like result, measurements should be "
+                    "included manually and an observable should not be "
+                    "used."
+                )
+            else:
+                # Using an observable but no type hinting
+                raise ValueError(
+                    "When using an observable, the return type must be "
+                    "specified on the user defined executor."
+                )
+        elif observable is None:
+            # Type hinted as DensityMatrixLike but no observable is set
+            if self._executor_return_type in DensityMatrixLike:
+                raise ValueError(
+                    "When using a density matrix like result, an observable "
+                    "is required."
```

Comment:
```suggestion
                    "When using a density matrix result, an observable "
                    "is required."
```

Reply:
I used your wording, just corrected the formatting in https://github.com/unitaryfund/mitiq/pull/2514/commits/19934a9d22d4ea381f6b7de7b113f5052b0a6ef8